flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/taskmanager/TaskManagerLogsHandlerTest.java

```diff
@@ -0,0 +1,102 @@
+ */
+
+package org.apache.flink.runtime.rest.handler.taskmanager;
+
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
+import org.apache.flink.runtime.concurrent.FutureUtils;
+import org.apache.flink.runtime.resourcemanager.exceptions.UnknownTaskExecutorException;
+import org.apache.flink.runtime.resourcemanager.utils.TestingResourceManagerGateway;
+import org.apache.flink.runtime.rest.handler.HandlerRequest;
+import org.apache.flink.runtime.rest.handler.HandlerRequestException;
+import org.apache.flink.runtime.rest.messages.EmptyRequestBody;
+import org.apache.flink.runtime.rest.messages.taskmanager.LogInfo;
+import org.apache.flink.runtime.rest.messages.taskmanager.LogsInfo;
+import org.apache.flink.runtime.rest.messages.taskmanager.TaskManagerIdPathParameter;
+import org.apache.flink.runtime.rest.messages.taskmanager.TaskManagerLogsHeaders;
+import org.apache.flink.runtime.rest.messages.taskmanager.TaskManagerMessageParameters;
+import org.apache.flink.runtime.testingUtils.TestingUtils;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for the {@link TaskManagerLogsHandler}.
+ */
+public class TaskManagerLogsHandlerTest extends TestLogger {
+
+	private static final ResourceID EXPECTED_TASK_MANAGER_ID = ResourceID.generate();
+
+	@Test
+	public void testGetTaskManagerLogsList() throws Exception {
+		final TestingResourceManagerGateway resourceManagerGateway = new TestingResourceManagerGateway();
+		final TaskManagerLogsHandler taskManagerLogsHandler = new TaskManagerLogsHandler(
+			() -> CompletableFuture.completedFuture(null),
+			TestingUtils.TIMEOUT(),
+			Collections.emptyMap(),
+			TaskManagerLogsHeaders.getInstance(),
+			() -> CompletableFuture.completedFuture(resourceManagerGateway));
+		final HandlerRequest<EmptyRequestBody, TaskManagerMessageParameters> handlerRequest = createRequest(EXPECTED_TASK_MANAGER_ID);
+		List<LogInfo> logsList = new ArrayList<>();
+		logsList.add(new LogInfo("taskmanager.log", 1024L));
+		logsList.add(new LogInfo("taskmanager.out", 1024L));
+		logsList.add(new LogInfo("taskmanager-2.out", 1024L));
+		resourceManagerGateway.setRequestTaskManagerLogListFunction(EXPECTED_TASK_MANAGER_ID -> CompletableFuture.completedFuture(logsList));
+		LogsInfo logsInfo = taskManagerLogsHandler.handleRequest(handlerRequest, resourceManagerGateway).get();
+		assertEquals(logsInfo.getLogInfos().size(), resourceManagerGateway.requestTaskManagerLogList(EXPECTED_TASK_MANAGER_ID, TestingUtils.TIMEOUT()).get().size());
+	}
+
+	@Test
+	public void testGetTaskManagerLogsListForUnknownTaskExecutorException() throws Exception {
+		final TestingResourceManagerGateway resourceManagerGateway = new TestingResourceManagerGateway();
```

Comment:
There is a bit of redundancy here. I would initialize the handler in a setUp method:

```
	@Before
	public void setUp() {
		resourceManagerGateway = new TestingResourceManagerGateway();
		taskManagerLogsHandler = new TaskManagerLogsHandler(
			() -> CompletableFuture.completedFuture(null),
			TestingUtils.TIMEOUT(),
			Collections.emptyMap(),
			TaskManagerLogsHeaders.getInstance(),
			() -> CompletableFuture.completedFuture(resourceManagerGateway));
	}
```